test/boost/frozen_mutation_test.cc

```diff
@@ -70,6 +71,33 @@ SEASTAR_TEST_CASE(test_writing_and_reading_gently) {
     });
 }
 
+// Called in a seastar thread
+template <typename FreezeFunc>
+requires std::same_as<std::invoke_result_t<FreezeFunc, const std::vector<mutation>&>, std::vector<frozen_mutation>>
+static void _test_freeze_unfreeze_vector(const std::vector<mutation>& mutations, FreezeFunc&& freeze_func) {
```

Comment:
Use std::function instead of instantiating the template multiple times.

Reply:
What's this leading underscore?

Reply:
It's a convention for a "private" function used in this file.
I borrowed that from `_test_freeze_unfreeze` that preexists in this test file.
That said, I don't feel strongly about it.

Reply:
`static`

Reply:
yes, it is.